### PR TITLE
[ci][automation/3] script to run state machine

### PR DESCRIPTION
**Pull request by can-anyscale**
A script to run CI state machine. Later on I'll create a buildkite job to run this nightly.

Test:
- CI

```
bazel run //ci/ray_ci/automation:state_machine_bot
[INFO 2024-01-11 23:55:46,018] state_machine_bot.py: 21  Starting state machine bot ...
```

### CODE REVIEW

File: ci/ray_ci/BUILD.bazel [hunk 1]
```diff
@@ -34,6 +34,12 @@ py_binary(
     deps = [":ray_ci_lib"],
 )
 
+py_binary(
+    name = "move_test_state",
+    srcs = ["move_test_state.py"],
+    deps = [":ray_ci_lib"],
```
Comment (aslonnie) [inline ci/ray_ci/BUILD.bazel:40]: just move this `py_binary` into the subdir, and declare rayci lib as deps?

don't include the script in the lib's source file list.

File: ci/ray_ci/move_test_state.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+from ci.ray_ci.automation.state_machine_bot import main
```
Comment (aslonnie) [inline ci/ray_ci/move_test_state.py:1]: just remove this wrapper. it is a script (with no test). you do not really need this wrapper.

**Comment by can-anyscale** [comment]
@aslonnie 's feedback

File: ci/ray_ci/BUILD.bazel [hunk 3]
```diff
@@ -11,7 +11,7 @@ py_library(
             "build_in_docker.py",
         ],
     ),
-    visibility = ["//visibility:private"],
+    visibility = ["//visibility:public"],
```
Comment (aslonnie) [inline ci/ray_ci/BUILD.bazel:14]: could you change this to limit to `ci/ray_ci`?

**Review by aslonnie — APPROVED** [review]

**Pull request merged**